README


                           The TauP Toolkit

            Flexible Seismic Travel-Time and Raypath Utilities
                              Version 2.0
                      http://www.seis.sc.edu/taup/

The TauP Toolkit is a seismic travel time calculator. In addition to travel
times, it can calculate derivative information such as ray paths through the
earth, pierce and turning points. It handles many types of velocity models and
can calculate times for virtually any seismic phase with a phase parser.
It is written in Java so it should run on any Java enabled platform.

The manual for the TauP Toolkit can be found in doc/taup.pdf. A copy of the TauP paper published in SRL can be found in doc/SRL_paper/taup_srl_with_figs.pdf.

Scripts for the various tool for both unix (sh) and windows (bat) are in the bin directory.

The website for the TauP Toolkit is:
http://www.seis.sc.edu/TauP

We use gradle to compile TauP. Should you wish to recompile, the gradlew wrapper will download gradle and compile. For example 'gradle/gradlew jar' will compile the java source and create a new jar. 'gradle eclipse' will generate an eclipse project files for importing. More information can be found at http://www.gradle.org.

If you like TauP and would like to cite it, please use the following:
Crotwell, H. P., T. J. Owens, and J. Ritsema (1999). The TauP Toolkit: Flexible seismic travel-time and ray-path utilities, Seismological Research Letters 70, 154–160.

Comments, criticism and random remarks may be sent to taup@seis.sc.edu.


### Rebuilding

You should not need to rebuild TauP unless you are trying to help debug, but if you do, you can build it using the Gradle wrapper script.

gradlew eB

will build TauP in build/explode.


docs/HISTORY

-------------------------------------------------------------------------------
                       Modifications to TauP
Vers  Date    Description
-------------------------------------------------------------------------------
2.0  July 11, 2011
      1) Code modernization
      2) Build with Gradle
      3) Add taup_console for jython integration
1.1.7-grh1 Feb. 25, 2007 G. Helffrich/U. Bristol
      1) Eliminate use of log4j package; only used one place in code for
	 diagnostic message.  Diagnostic replaced with output under verbose
	 control.
      2) Add Makefile to main directory, change source directory Makefile to
	 invoke java compiler javac properly.
      3) Add -expert mode to TauP_{Time,Pierce,Path} to allow sources in the
	 core.  This is to provide a straightforward way to do point
	 scattering calculations.
      4) Fix typo in PREM model depths.
      5) Improve label positions on reverse branches in TauP_Curve.
      6) Add fortran call interface to subroutine call interface to TauP_Time
      7) Compile and install dynamically loadable library for interface routines
1.1.7  Apr 22, 2006
      1) Dumb rerelease due to compiling under java1.5
1.1.6  Apr 21, 2006
      1) Fix bug with corrupted data in little-endian sac files due to NaN translation by Java
1.1.5  Aug 3, 2005
      1) Fixed bug with pierce points and upgoing Pdiff leg
      2) Autodetect byte swapping in SAC files
      3) Put ray param into user headers in TauP_SetSac
1.1.4  Apr 5, 2002
      1) Added fix for models with S vel in Mantle higher than P vel in core, which happens in some Mars models. (Jadhav)
      2) Added time to both taup_pierce and taup_path (Rivera)
1.1.3  Jul 18, 2001
      1) Fixed bug with taup_path always using iasp91 (Pulliam)
1.1.2  Feb 9, 2001
      1) Allowed TauP GUI to be called from other tools
      2) Fixed bug in TauP_Path with outFile being null
      3) Added model text files to distribution(Ritsema)
1.1.1  April 19, 2000
      1) fixed problem with p^mP type phases for deep sources (Owens,Zandt,Ritsema)
      2) added getSlownessModel and getVelocityModel to TauModel and 
                getVelocityModel to SlownessModel
1.1    February 2, 2000
      1) several fixes to documentation (Snoke)
      2) now GPL'ed
      3) with source code
      4) add a GUI for times, pierce points and paths, not very advanced but 
               interesting
1.0.9 12/15/99
      1) fixed bug with taup_setsac not finding any arrivals
1.0.8 11/29/99
		1) fixed path bug for S at 180
		2) fixed SphericalCoords.lonFor() to be between -180 and 180
		3) changed PropertyLoader and TauModelLoader to use					getResourceAsStream to work with JDK1.2 where the				jar may not be in the classpath
		4) fixed bug with loading models from user generated jar files,
			manual said Models/modelname but code had 
			StdModels/modelname
		5) small changes and updates to manual, including adding
			license info
		6) added java2 install instructions and Mac and Windows
			 specific instructions
		7) fixed problems with jacl scripts
                8) added configurable precision for outputs, see standard 
                        properties
                9) redid formating of manual, used TimesRoman and pdf is now
                        usable, plus has hyperlinked table of contents, etc
                10) fixed null pointer bug with -help in TauP_Time
1.0.7 2/9/99   
      1) forgot to flush BufferedOutputWriter, resulting in no output at all
1.0.6	2/8/99 	1) fixed bug with very small layers and power law, now very 
		    small (< 1 micron) layers are just 0 time and distance
		    (Firpo)
		2) fixed bug with pierce depths being printed below turning 
		    depth (Ritsema)
		3) changed from printResult(StringWriter) to 
		    printResult(Writer) to fix out of memory bug
		4) added radius of earth to verbose option in 
		    taup_create (Owens)
		5) fixed bug with power law depth finder giving depths just
		    outside the layer
		6) general cleanup of source code
1.0.5  12/3/98 1)Speed improvement in TauP_SetSac, replaced readData with new
                   method directly calling intbitstofloat.
               2)fixed bug in TauP_Time, it was not depth correcting in some 
                   cases
1.0.4  11/23/98 Removed unneeded printlns from TauP_SetSac
1.0.3  11/17/98 Bug fix
               1) fixed (hopefully) a numerical instability in SlownessLayer
                   due to using the power law v=Ar^B in very thin layers
1.0.1  10/28/98 Bug fixes
               1) fixed bug in formater that rounded incorectly, 14.99 would
                   print as 14.0 when formatted %5.1f
               2) fixed taup_table bug when Pdiff was only arrival and distance
                   was greater than 105 degrees 
               3) fixed bug with depth of 15km in iasp91, indexP and indexS
                   were being incorrectly swapped

1.0    9/9/98  Public release
               1) added purist's phase names
               2) added taup_table tool
               3) fixed bug with taup_path not going all the way to the surface
               4) user's guide updated
               5) added get methods to Arrival and made attributes protected
               6) made time and distance calculations for layers more effecient
               7) added coarsely sampled qdt model for quick and dirty times
               8) several minor fixes

0.96.3 8/26/98 1) fixed taup.path to taup.model.path changes that were forgotten

0.96.2 8/25/98 1) added ScS to tts+

0.96.1 8/19/98 1) added check for max error due to linear interpolation


src/edu/sc/seis/TauP/TauP.java
package edu.sc.seis.TauP;

import javax.swing.DefaultListModel;
import javax.swing.JTable;
import javax.swing.event.ListSelectionEvent;
import javax.swing.event.ListSelectionListener;

/*
 * The TauP Toolkit: Flexible Seismic Travel-Time and Raypath Utilities.
 * Copyright (C) 1998-2000 University of South Carolina
 * 
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later
 * version.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 * 
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
 * Place - Suite 330, Boston, MA 02111-1307, USA.
 * 
 * The current version can be found at <A
 * HREF="www.seis.sc.edu">http://www.seis.sc.edu</A>
 * 
 * Bug reports and comments should be directed to H. Philip Crotwell,
 * crotwell@seis.sc.edu or Tom Owens, owens@seis.sc.edu
 * 
 */
/**
 * 
 * @author H. Philip Crotwell
 * @version 1.1.3 Wed Jul 18 15:00:35 GMT 2001
 * 
 * 
 * 
 */
public class TauP extends javax.swing.JFrame {

    /** Initializes the Form */
    protected double distance = 0;

    protected javax.swing.JTable timeTable, pierceTable;

    protected ArrivalTableModel arrivalTM;

    PierceTableModel pierceTableModel;

    protected PathPlot pathPlotArea;

    public TauP() {
        super("The TauP Toolkit");
        initComponents();
        pack();
        try {
            timeTool = new TauP_Time("iasp91");
            timeTool.init();
            pierceTool = new TauP_Pierce(timeTool.getTauModel());
            pierceTool.init();
            pathTool = new TauP_Path(timeTool.getTauModel());
            pathTool.init();
            phaseDialog = new PhaseDialog(this, false);
            phaseDialog.setPhases(timeTool.getPhaseNames());
            pathPlotArea = new PathPlot();
            pathPlotArea.setPreferredSize(new java.awt.Dimension(500, 300));
            pathPlotArea.setMinimumSize(new java.awt.Dimension(500, 200));
            pathPanel.add(pathPlotArea, java.awt.BorderLayout.CENTER);
        } catch(TauModelException e) {
            System.out.println(e);
        } catch(java.io.IOException e) {
            System.out.println(e);
        }
        arrivalTM = new ArrivalTableModel();
        timeTable = new javax.swing.JTable(arrivalTM);
        javax.swing.JScrollPane timeTableScrollPane = new javax.swing.JScrollPane(timeTable);
        timeTableScrollPane.setPreferredSize(new java.awt.Dimension(500, 300));
        timeTableScrollPane.setMinimumSize(new java.awt.Dimension(500, 200));
        timePanel.add(timeTableScrollPane, java.awt.BorderLayout.CENTER);
        modelComboBox.addItem("iasp91");
        modelComboBox.addItem("prem");
        modelComboBox.addItem("qdt");
        setSize(new java.awt.Dimension(500, 300));
        pierceTableModel = new PierceTableModel();
        pierceTable = new JTable(pierceTableModel);
        pierceArrivalList.addListSelectionListener(new ListSelectionListener() {

            public void valueChanged(ListSelectionEvent e) {
                if(!e.getValueIsAdjusting()) {
                    pierceListValueChanged(e);
                }
            }
        });
        jScrollPane2.setViewportView(pierceTable);
        pierceModel = new DefaultListModel();
        pierceArrivalList.setModel(pierceModel);
        // path list
        jList1.setModel(pierceModel);
        jList1.setSelectionModel(pierceArrivalList.getSelectionModel());
        validate();
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the FormEditor.
     */
    private void initComponents() {// GEN-BEGIN:initComponents
        // This code was developed using a non-commercially licensed version of
        // NetBeans Developer 2.x.
        // For details, see http://www.netbeans.com/non_commercial.html
        addWindowListener(new java.awt.event.WindowAdapter() {

            public void windowClosing(java.awt.event.WindowEvent evt) {
                exitForm(evt);
            }
        });
        getContentPane().setLayout(new java.awt.BorderLayout());
        jTabbedPane1 = new javax.swing.JTabbedPane();
        jTabbedPane1.setMinimumSize(new java.awt.Dimension(200, 200));
        timePanel = new javax.swing.JPanel();
        timePanel.setToolTipText("travel times");
        timePanel.setName("Time");
        timePanel.setLayout(new java.awt.BorderLayout());
        jTabbedPane1.addTab("Time", null, timePanel, "travel times");
        piercePanel = new javax.swing.JPanel();
        piercePanel.setToolTipText("pierce points");
        piercePanel.setName("Pierce");
        piercePanel.setLayout(new java.awt.BorderLayout());
        jScrollPane1 = new javax.swing.JScrollPane();
        pierceArrivalList = new javax.swing.JList();
        pierceArrivalList.setPreferredSize(new java.awt.Dimension(60, 60));
        pierceArrivalList.setSelectionMode(javax.swing.ListSelectionModel.SINGLE_SELECTION);
        pierceArrivalList.setMinimumSize(new java.awt.Dimension(60, 60));
        jScrollPane1.add(pierceArrivalList);
        jScrollPane1.setViewportView(pierceArrivalList);
        piercePanel.add(jScrollPane1, "West");
        jScrollPane2 = new javax.swing.JScrollPane();
        piercePanel.add(jScrollPane2, "Center");
        jTabbedPane1.addTab("Pierce", piercePanel);
        pathPanel = new javax.swing.JPanel();
        pathPanel.setToolTipText("path points");
        pathPanel.setName("Path Points");
        pathPanel.setLayout(new java.awt.BorderLayout());
        jScrollPane3 = new javax.swing.JScrollPane();
        jList1 = new javax.swing.JList();
        jList1.setPreferredSize(new java.awt.Dimension(60, 60));
        jList1.setMinimumSize(new java.awt.Dimension(60, 60));
        jScrollPane3.setViewportView(jList1);
        pathPanel.add(jScrollPane3, "West");
        jTabbedPane1.addTab("Path", pathPanel);
        getContentPane().add(jTabbedPane1, "Center");
        jPanel1 = new javax.swing.JPanel();
        jPanel1.setLayout(new java.awt.GridBagLayout());
        java.awt.GridBagConstraints gridBagConstraints1;
        jLabel1 = new javax.swing.JLabel();
        jLabel1.setText("Model");
        jLabel1.setName("modelLabel");
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 0;
        gridBagConstraints1.gridy = 0;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        jPanel1.add(jLabel1, gridBagConstraints1);
        modelComboBox = new javax.swing.JComboBox();
        modelComboBox.setToolTipText("choose a model");
        modelComboBox.setActionCommand("modelChanged");
        modelComboBox.setName("modelMenu");
        modelComboBox.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                chooseModelActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 1;
        gridBagConstraints1.gridy = 0;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        jPanel1.add(modelComboBox, gridBagConstraints1);
        jLabel2 = new javax.swing.JLabel();
        jLabel2.setText("Distance (deg)");
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 2;
        gridBagConstraints1.gridy = 0;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        gridBagConstraints1.insets = new java.awt.Insets(0, 10, 0, 0);
        gridBagConstraints1.anchor = java.awt.GridBagConstraints.EAST;
        jPanel1.add(jLabel2, gridBagConstraints1);
        distanceTextField = new javax.swing.JTextField();
        distanceTextField.setToolTipText("distance in degrees");
        distanceTextField.setPreferredSize(new java.awt.Dimension(100, 21));
        distanceTextField.setMinimumSize(new java.awt.Dimension(100, 21));
        distanceTextField.setText("50.0");
        distanceTextField.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                distanceActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 3;
        gridBagConstraints1.gridy = 0;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        jPanel1.add(distanceTextField, gridBagConstraints1);
        jLabel3 = new javax.swing.JLabel();
        jLabel3.setText("Depth");
        jLabel3.setHorizontalAlignment(javax.swing.SwingConstants.RIGHT);
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 2;
        gridBagConstraints1.gridy = 1;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        gridBagConstraints1.insets = new java.awt.Insets(0, 10, 0, 0);
        gridBagConstraints1.anchor = java.awt.GridBagConstraints.EAST;
        jPanel1.add(jLabel3, gridBagConstraints1);
        depthTextField = new javax.swing.JTextField();
        depthTextField.setToolTipText("source depth in km");
        depthTextField.setPreferredSize(new java.awt.Dimension(100, 21));
        depthTextField.setMinimumSize(new java.awt.Dimension(100, 21));
        depthTextField.setText("0.0");
        depthTextField.setName("depthTextField");
        depthTextField.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                depthActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 3;
        gridBagConstraints1.gridy = 1;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        jPanel1.add(depthTextField, gridBagConstraints1);
        phaseChooserButton = new javax.swing.JButton();
        phaseChooserButton.setToolTipText("show phase chooser");
        phaseChooserButton.setText("Phases");
        phaseChooserButton.setActionCommand("showPhases");
        phaseChooserButton.setLabel("Phases");
        phaseChooserButton.setName("phaseButton");
        phaseChooserButton.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                phaseChooserActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 1;
        gridBagConstraints1.gridy = 1;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        jPanel1.add(phaseChooserButton, gridBagConstraints1);
        calculateButton = new javax.swing.JButton();
        calculateButton.setToolTipText("calculate times");
        calculateButton.setText("Calculate");
        calculateButton.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                calculateActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 4;
        gridBagConstraints1.gridy = 0;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        gridBagConstraints1.insets = new java.awt.Insets(0, 20, 0, 0);
        jPanel1.add(calculateButton, gridBagConstraints1);
        jButton1 = new javax.swing.JButton();
        jButton1.setText("Quit");
        jButton1.setActionCommand("quit");
        jButton1.setLabel("Quit");
        jButton1.addActionListener(new java.awt.event.ActionListener() {

            public void actionPerformed(java.awt.event.ActionEvent evt) {
                quitActionPerformed(evt);
            }
        });
        gridBagConstraints1 = new java.awt.GridBagConstraints();
        gridBagConstraints1.gridx = 4;
        gridBagConstraints1.gridy = 1;
        gridBagConstraints1.fill = java.awt.GridBagConstraints.HORIZONTAL;
        gridBagConstraints1.insets = new java.awt.Insets(0, 20, 0, 0);
        jPanel1.add(jButton1, gridBagConstraints1);
        getContentPane().add(jPanel1, "South");
    }// GEN-END:initComponents

    protected boolean quitExits = false;

    /**
     * Get the value of quitExits.
     * 
     * @return Value of quitExits.
     */
    public boolean getQuitExits() {
        return quitExits;
    }

    /**
     * Set the value of quitExits.
     * 
     * @param v
     *            Value to assign to quitExits.
     */
    public void setQuitExits(boolean v) {
        this.quitExits = v;
    }

    private void phaseChooserActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_phaseChooserActionPerformed
        // Add your handling code here:
        phaseDialog.setVisible(true);
    }// GEN-LAST:event_phaseChooserActionPerformed

    private void quitActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_quitActionPerformed
        // Add your handling code here:
        if(getQuitExits()) {
            System.exit(0);
        } else {
            dispose();
        }
    }// GEN-LAST:event_quitActionPerformed

    private void calculateActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_calculateActionPerformed
        // Add your handling code here:
        try {
            timeTool.setSourceDepth(Double.valueOf(depthTextField.getText())
                    .doubleValue());
            timeTool.clearPhaseNames();
            timeTool.setPhaseNames(phaseDialog.getPhases());
            setDistance(Double.valueOf(distanceTextField.getText())
                    .doubleValue());
            timeTool.calculate(distance);
            // System.out.println("Done calculating time"+distance+"
            // "+timeTool.getNumArrivals()
            // +" "+timeTool.getPhaseNameString()+" "+timeTool.getTauModelName()
            // );
            pierceTool.setSourceDepth(Double.valueOf(depthTextField.getText())
                    .doubleValue());
            pierceTool.clearPhaseNames();
            pierceTool.setPhaseNames(phaseDialog.getPhases());
            pierceTool.calculate(distance);
            pathTool.setSourceDepth(Double.valueOf(depthTextField.getText())
                    .doubleValue());
            pathTool.clearPhaseNames();
            pathTool.setPhaseNames(phaseDialog.getPhases());
            pathTool.calculate(distance);
            showResults();
        } catch(Exception e) {
            System.out.println("exception in calculate " + e);
            e.printStackTrace();
        }
    }// GEN-LAST:event_calculateActionPerformed

    private void distanceActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_distanceActionPerformed
        // Add your handling code here:
        setDistance(Double.valueOf(evt.getActionCommand()).doubleValue());
        calculateActionPerformed(evt);
    }// GEN-LAST:event_distanceActionPerformed

    private void depthActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_distanceActionPerformed
        // Add your handling code here:
        setDepth(Double.valueOf(evt.getActionCommand()).doubleValue());
        calculateActionPerformed(evt);
    }// GEN-LAST:event_distanceActionPerformed

    private void chooseModelActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_chooseModelActionPerformed
        // Add your handling code here:
        setModel((String)(modelComboBox.getSelectedItem()));
    }// GEN-LAST:event_chooseModelActionPerformed

    private void pierceListValueChanged(ListSelectionEvent e) {
        pierceTableModel.setSelectedIndex(pierceArrivalList.getSelectedIndex());
        pathPlotArea.setSelectedIndex(pierceArrivalList.getSelectedIndex());
    }

    /** Exit the Application */
    private void exitForm(java.awt.event.WindowEvent evt) {// GEN-FIRST:event_exitForm
        if(getQuitExits()) {
            System.exit(0);
        } else {
            dispose();
        }
    }// GEN-LAST:event_exitForm

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JTabbedPane jTabbedPane1;

    private javax.swing.JPanel jPanel1;

    private javax.swing.JPanel timePanel;

    private javax.swing.JPanel piercePanel;

    private javax.swing.JPanel pathPanel;

    private javax.swing.JScrollPane jScrollPane1;

    private javax.swing.JScrollPane jScrollPane2;

    private javax.swing.JList pierceArrivalList;

    private javax.swing.JLabel jLabel1;

    private javax.swing.JComboBox modelComboBox;

    private javax.swing.JLabel jLabel2;

    private javax.swing.JTextField distanceTextField;

    private javax.swing.JLabel jLabel3;

    private javax.swing.JTextField depthTextField;

    private javax.swing.JButton phaseChooserButton;

    private javax.swing.JButton calculateButton;

    private javax.swing.JButton jButton1;

    private javax.swing.JScrollPane jScrollPane3;

    private javax.swing.JList jList1;

    // End of variables declaration//GEN-END:variables
    private TauP_Time timeTool;

    private TauP_Pierce pierceTool;

    private TauP_Path pathTool;

    private PhaseDialog phaseDialog;

    private DefaultListModel pierceModel;

    protected void setModel(String modelName) {
        try {
            timeTool.loadTauModel(modelName);
            pierceTool.setTauModel(timeTool.getTauModel());
            pathTool.setTauModel(timeTool.getTauModel());
        } catch(Exception e) {
            System.out.println(e);
        }
    }

    protected void setDepth(double depth) {
        timeTool.setSourceDepth(depth);
    }

    protected void setDistance(double distance) {
        this.distance = distance;
    }

    protected void showResults() {
        arrivalTM.setArrivals(timeTool.getArrivals());
        pierceTableModel.setArrivals(pierceTool.getArrivals());
        pierceModel.removeAllElements();
        for(int i = 0; i < pierceTool.getNumArrivals(); i++) {
            pierceModel.addElement(pierceTool.getArrival(i).getName());
        }
        pierceArrivalList.setSelectedIndex(0);
        pathPlotArea.setTauModel(pathTool.getTauModel());
        pathPlotArea.removeAllElements();
        for(int i = 0; i < pathTool.getNumArrivals(); i++) {
            pathPlotArea.addElement(pathTool.getArrival(i));
        }
        pathPlotArea.repaint();
    }

    public static void main(java.lang.String[] args) {
        TauP t = new TauP();
        t.setQuitExits(true);
        t.show();
    }
}
